src/AutoMapper/TypeMap.cs

```diff
@@ -281,6 +283,8 @@ public void Seal(IConfigurationProvider configurationProvider)
                     .OrderBy(map => map.MappingOrder).ToArray();
 
             MapExpression = CreateMapperLambda(configurationProvider, null);
+
+            ((FeatureCollection)Features).Seal(configurationProvider);
```

Comment:
Maybe just drop all the interfaces and use the classes instead?

Reply:
I don't see the need for the interfaces for a lot of places for this feature, but AM is using interfaces for most of the classes I added functionality into so tried to only follow the pattern. Should I clean the interfaces that not is needed away?

Reply:
That's to better separate public API from implementation. But given the purpose of this feature, I think classes are just fine. cc @jbogard